### PR TITLE
feat: add provider configuration for ofrep

### PR DESCRIPTION
**Pull request by thepabloaguilar**
This PR can be considered the very first step towards the implementation of OpenFeature Remote Evaluation Protocol!

It's the implementation of an optional endpoint used by the protocol, I've decided to implement this endpoint first instead of the others (which actually are mandatory) because its simplicity and we can discuss some things here better.

The first thing you may ask is about the endpoint route, which is: `/ofrep/*/ofrep/v1/configuration`
Why we have `ofrep` twice? The answer is, because OFREP defines all the routes as follow:
* /ofrep/v1/evaluate/flags/{key}
* /ofrep/v1/evaluate/flags
* /ofrep/v1/configuration

And as you can see there's a important part missing in those, the __namespace__ flipt supports! So the strategy we can use is always prefix with `/ofrep/{namespace}` and when instantiating the ofrep client just give the address as follow: `http://my.flipt/ofrep/my-namespace`

Refs #2980

### CODE REVIEW

File: rpc/flipt/flipt.yaml [hunk 1]
```diff
@@ -323,3 +323,9 @@ http:
   # method: evaluation count
   - selector: flipt.analytics.AnalyticsService.GetFlagEvaluationsCount
     get: /internal/v1/analytics/namespaces/{namespace_key}/flags/{flag_key}
+
+  # Open Feature Remote Evaluation Protocol (OFREP)
+  #
+  # method: provider configuration
+  - selector: flipt.ofrep.OFREPService.GetProviderConfiguration
+    get: /ofrep/*/ofrep/v1/configuration
```
Comment (GeorgeMac) [inline rpc/flipt/flipt.yaml:331]:
```suggestion
    get: /ofrep/{namespace_key}/ofrep/v1/configuration
```

Our SDK client generator doesn't really know how to handle this wildcard. So at the moment, it is generating a request to this path as you have it here with an explicit `*` in it. If we made this explicitly the namespace key and added that field to the `GetProviderConfigurationRequest` (even though we're not going to use it), that will ensure the client gets generated in such a way that it doesn't render a path with `*` in it.

Or maybe it doesn't really matter as maybe no one will use the OFREP section of our Go SDK.
We could actually use our magic comment that stops it from getting generated altogether.

File: sdk/go/http/ofrep.sdk.gen.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+type OFREPServiceClient struct {
+	client *http.Client
+	addr   string
+}
+
+func (x *OFREPServiceClient) GetProviderConfiguration(ctx context.Context, v *ofrep.GetProviderConfigurationRequest, _ ...grpc.CallOption) (*ofrep.GetProviderConfigurationResponse, error) {
+	var body io.Reader
+	var values url.Values
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, x.addr+"/ofrep/*/ofrep/v1/configuration", body)
```
Comment (GeorgeMac) [inline sdk/go/http/ofrep.sdk.gen.go:23]: This is the generated path I was referring to. Could either make it a real namespace parameter or we could skip this generation altogether.
https://github.com/flipt-io/flipt/blob/50af0bc4d6f6e94a58bdfd0108703e82095acd15/rpc/flipt/evaluation/evaluation.proto#L201-L204

**Review by GeorgeMac — COMMENTED** [review]
This is looking great @thepabloaguilar. Thanks for taking it on.
One thought on here regarding our SDK generator and that path.
Might be worth skipping the SDK generation for the OFREP service as we're probably never going to use it.

**Comment by markphelps** [comment]
Hey @thepabloaguilar thanks for getting this work started!! Very much appreciated!

I actually asked in the OpenFeature Slack channel a couple months ago about the best way to support namespaces in our case, and they suggested we use the custom header support that OFREP clients already provide.

Here's the link to the slack thread (you need to be in the CNFC slack first I think): https://cloud-native.slack.com/archives/C066A48LK35/p1714414882954439

So in our case we would do something like this (using Go OFREP provider as an example):

```go
provider := ofrep.NewProvider(
    "http://localhost:8080",
    ofrep.WithHeaderProvider(func() (name string, value string) {
      return "X-Flipt-Namespace", "foo"
    })
)
```

Then in our server side impl we would need to pull out this header value to get the correct namespace

File: sdk/go/http/ofrep.sdk.gen.go (hunk 2 above)
Comment (thepabloaguilar) [reply sdk/go/http/ofrep.sdk.gen.go:23]: Sure, I'll put the ignore comment and by the @markphelps message I guess we can just get ride of `namespace` thing

**Comment by thepabloaguilar** [comment]
Thanks for the comment @markphelps! I thought about using the header strategy first but I was like: Idk if it's correct to put a specific thing in something which is trying to generalize the interactions. But I'd stick with that, will add the header handling

**Comment by thepabloaguilar** [comment]
I've tested locally and it seems to be compliant with the OpenAPI Spec:
```sh
curl localhost:8080/ofrep/v1/configuration
{"name":"flipt","capabilities":{"cacheInvalidation":{"polling":{"enabled":false,"minPollingIntervalMs":"0"}},"flagEvaluation":{"supportedTypes":["string","boolean"]}}}
```

Prettyfied:
```json
{
  "name": "flipt",
  "capabilities": {
    "cacheInvalidation": {
      "polling": {
        "enabled": false,
        "minPollingIntervalMs": "0"
      }
    },
    "flagEvaluation": {
      "supportedTypes": [
        "string",
        "boolean"
      ]
    }
  }
}
```

**Comment by erka** [comment]
> I've tested locally and it seems to be compliant with the OpenAPI Spec:
> 
> ```shell
> curl localhost:8080/ofrep/v1/configuration
> {"name":"flipt","capabilities":{"cacheInvalidation":{"polling":{"enabled":false,"minPollingIntervalMs":"0"}},"flagEvaluation":{"supportedTypes":["string","boolean"]}}}
> ```

Hi @thepabloaguilar. Great work of OFREP!

There is one small thing that isn't in complaint with OFREP OpenAPI Spec:

```yaml
minPollingIntervalMs:
  type: number
     description: |
         Minimum polling interval (in millisecond) supported by the flag management system.
          The provider should ensure not to set any polling value under this minimum.
      examples:
          - 60000
```

In our case, `minPollingIntervalMs` is a string not a number. This discrepancy arises because in gRPC/protobuf, `int64` is encoded as a `string`. To be fully compliant with the OFREP spec, we could change `minPollingIntervalMs` to something like `uint32`. WDYT?

**Comment by thepabloaguilar** [comment]
Sure @erka, I've just pushed the changes! But AFAIK even as a string it's a valid _number_ for the JSON parsers

**Comment by erka** [comment]
> Sure @erka, I've just pushed the changes! But AFAIK even as a string it's a valid _number_ for the JSON parsers

@thepabloaguilar  Unfortunately, Go doesn't 
```go
var r map[string]int
err := json.Unmarshal([]byte(`{"a":"1"}`), &r)
fmt.Println(r, err)
// Output:
// map[a:0] json: cannot unmarshal string into Go value of type int
```

**Comment by erka** [comment]
@thepabloaguilar One downside with `mage proto` is that it doesn't remove previously generated files when proto files change.  We have to remove`sdk/go/http/ofrep.sdk.gen.go` and `sdk/go/ofrep.sdk.gen.go` manually :(

**Comment by thepabloaguilar** [comment]
@erka done!

**Review by erka — APPROVED** [review]
@thepabloaguilar Excellent beginning on the new feature!

**Review by markphelps — APPROVED** [review]
looks great to me @thepabloaguilar !! Thank you for kicking off this work 🚀

**Comment by thepabloaguilar** [comment]
I'm already working on the evaluate endpoint!

**Pull request merged**